### DISCUSSION TITLE
How to get both hand's joint's rotation from Hololens2 through TrackedHandJoint?

### DISCUSSION THREAD

**Discussion by mcoy001** [microsoft/MixedRealityToolkit-Unity#10003] [Q&A] [answered]
Hi, I'm trying to get both hand joint's rotation on hololens2.
I tried TrackedHandJoint with code from the hand tracking page
(https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/input/hand-tracking?view=mrtkunity-2021-05):
if (eventData.InputData.TryGetValue(TrackedHandJoint.IndexTip, out fingerTip))
{
    indextip_rotation = fingerTip.Rotation;
    text += indextip_rotation + "\n";
}
it works with one hand, but the data will be constantly updated by both hand if two hands show up.
For my current case I want to keep getting data from the right hand's index finger, while the left hand is still in the scene doing some gesture. Does anyone know how to do it?

**Comment by mcoy001** [comment, marked as answer]
I figured out  the method I was using from the demo will just get handjoint data from any matched ones, it's listening from both hands afterall
.
I just need to use HandJointService.RequestJointTransform which is exactly what I need to get the joint data I wanted.

**Comment by CDiaz-MS** [comment]
You could also use the HandJointUtils.TryGetJointPose to get the pose of specific joints for the specified hand:
```
MixedRealityPose rightIndexJoint;
HandJointUtils.TryGetJointPose(TrackedHandJoint.IndexTip, Handedness.Right, out rightIndexJoint);
```

**Reply by mcoy001** [reply]
thanks! Also I think it's more straight forward to use HandJointUtils for my case.
